lib/charms/loki_k8s/v0/log_proxy.py

```diff
@@ -586,26 +599,48 @@ def _scrape_configs(self) -> dict:
             A dict representing the `scrape_configs` section.
         """
         # TODO: use the JujuTopology object
-        config = {
-            "targets": ["localhost"],
-            "labels": {
-                "job": "juju_{}_{}_{}".format(
-                    self._charm.model.name,
-                    self._charm.model.uuid,
-                    self._charm.model.app.name,
-                ),
-                "__path__": "",
-            },
+        job_name = "juju_{}_{}_{}".format(
+            self._charm.model.name, self._charm.model.uuid, self._charm.model.app.name
+        )
+        common_labels = {
+            "juju_application": self._charm.app.name,
+            "juju_model": self._charm.model.name,
+            "juju_model_uuid": self._charm.model.uuid,
+            "juju_charm": self._charm.meta.name,
         }
 
-        return {
-            "scrape_configs": [
-                {
-                    "job_name": "system",
-                    "static_configs": self._generate_static_configs(config),
-                }
-            ]
+        scrape_configs = []
+
+        # Files config
+        labels = deepcopy(common_labels)
+        labels.update(
+            {
+                "job": job_name,
+                "__path__": "",
+            }
+        )
+        config = {"targets": ["localhost"], "labels": labels}
+        scrape_config = {
+            "job_name": "system",
+            "static_configs": self._generate_static_configs(config),
         }
+        scrape_configs.append(scrape_config)
+
+        # Syslog config
+        if self._is_syslog:
+            syslog_labels = deepcopy(common_labels)
+            syslog_labels.update({"job": "syslog_{}".format(job_name)})
+            syslog_config = {
+                "job_name": "syslog",
+                "syslog": {
+                    "listen_address": "127.0.0.1:{}".format(self._syslog_port),
+                    "label_structured_data": True,
+                    "labels": syslog_labels,
+                },
+            }
+            scrape_configs.append(syslog_config)  # type: ignore
+
+        return {"scrape_configs": scrape_configs}
```

Comment:
Consider updating the type hint accordingly, `Dict[List[dict]]`

Reply:
I don't understand where I would put this type hint

Reply:
Function definition currently has just dict.

Reply:
-1 from me here. If we're going to try to "properly" type it, we'd want to define a real type above, since `Dict[List[dict]]` isn't arbitrarily any clearer than just `dict` or `Dict[List]`. I'd go for  a simple `dict` for clarify rather than a middle ground which only partially expresses the datastructure, or define a "real" type which actually expresses the whole thing.

`Dict[List[Dict[str, Union[str, Dict[str, Union[str, List[str]]]]]]` gets unbearable really fast.

Reply:
`Dict[List[dict]]` is short and sweet to my eyes, and gives way more insight than `dict`, and very quickly so, into the type.
cf. [tradeoff](https://www.merriam-webster.com/dictionary/trade-off)

Reply:
This does not fix the need for `# type: ignore` so I see no point.